### ISSUE TITLE
NAS mounting NFS/CIFS

### ISSUE THREAD

**Issue by snowtr** [bbilly1/tubearchivist#31]
In my case, I have this running on a virtual server of Ubuntu Server 18.04 LTS and wish to use the VM to just run the tool while the actual content is stored on my NAS.

I can connect to my NAS through NFS or CIFS but prefer NFS for myself.

I have tried mounting the NFS share as /mnt/youtube and then creating a soft link to /youtube but that results in the docker making the media folder not save to the NAS.

I've also tried modifying the docker-compose.yml file to add the NFS mount to the docker image directly and have it mount the NAS path to ./volumes/tubearchivist/media but that also resulted in the content there being saved but not appearing in the NAS.

What can be done?

**Comment by bbilly1** [comment]
So **/youtube** is the path _inside_ the container. That has nothing to do with your host system. Docker doesn't care about the file protocol how you mount the NAS into the VM, that doesn't matter, well as long as the permissions are set correctly...

So if I'm understanding you correctly, you want to mount the NAS folder into the VM, you can do that wherever you'd like in the filetree. Then just point the volume to the correct path, so in your example, the media folder in the /mnt/youtube, for a volume you'd use that:
```
/mnt/youtube:/youtube
```
Then whatever is before the ':' is where the folder goes in the host system, and anything after ':' is the folder inside the container. Inside the container needs to be **/youtube** but on your host system, it can be whatever you want. 

Of course docker has excellent [documentation](https://docs.docker.com/storage/volumes/) about how volumes work.

**Comment by snowtr** [comment]
Right, now that I went through over a dozen articles and the documentation a few times, I figured out the order of that in the yml file.

It seems to be working now but I still suggest that you update the readme file to include details for those who don't know Docker.

**Comment by bbilly1** [comment]
That is beyond the scope of this project, I don't want to duplicate the docker documentation here. Additionally to the official documentation there are many very helpful "docker getting started" guides out there explaining basic docker concepts better than I ever could. Self hosting requires learning about the tools you are using. Keep it up, learning docker is totally worth it.

**Comment by fake-name** [comment]
> Self hosting requires learning about the tools you are using. Keep it up, learning docker is totally worth it.

So you need to learn java to use elasticsearch? And python just to _use_ tubearchivist? Or C/C++ to use windows (or linux?)

I agree that doing weird non-standard things isn't something that needs to be documented, but setting where downloads should go is so fundamental to using this thing that having a simple note for amateurs to understand how to configure things does not seem ridiculous. 

I managed to read the contents at `https://docs.tubearchivist.com/installation/docker-compose/#configuring-tubearchivist` without figuring out that `/youtube` wasn't the correct directory path myself. Does that mean I'm dumb? Probably, but if several people have made the same mistake (OP and I), it might be a decent signal that the docs could be improved.

**Comment by bbilly1** [comment]
if you see a way to jmprov ethe docs, please do, it's all open source.

I just don't want to end up duplicating dockers docs, or all these countles blog posts, they are a perfectly good resources to learn the basics of docker.
